test/Microsoft.AspNetCore.Routing.Tests/Matching/DfaMatcherTest.cs

```diff
@@ -106,6 +106,45 @@ public async Task MatchAsync_InvalidRouteConstraint_NoEndpointMatched()
             Assert.Null(context.Endpoint);
         }
 
+        [Fact]
+        public async Task MatchAsync_ParameterTransformer_EndpointMatched()
```

Comment:
Needs more integration tests that actually do URL matching and link generation with required values